Engine/Component/ComponentBillboard.cpp

```diff
@@ -0,0 +1,235 @@
+#include "ComponentBillboard.h"
+
+#include "Main/Application.h"
+#include "Module/ModuleProgram.h"
+#include "Module/ModuleResourceManager.h"
+#include "Module/ModuleRender.h"
+#include "Module/ModuleTime.h"
+
+#include "ResourceManagement/ResourcesDB/CoreResources.h"
+
+ComponentBillboard::ComponentBillboard() : Component(nullptr, ComponentType::BILLBOARD)
+{
+	InitData();
+}
+
+ComponentBillboard::ComponentBillboard(GameObject * _owner) : Component(owner, ComponentType::BILLBOARD)
+{
+	InitData();
+}
+
+ComponentBillboard::~ComponentBillboard()
+{
+	if (vbo != 0)
+	{
+		glDeleteBuffers(1, &vbo);
+		glDeleteBuffers(1, &ebo);
+		glDeleteVertexArrays(1, &vao);
+	}
+}
+
+void ComponentBillboard::InitData()
+{
+	ChangeTexture(static_cast<uint32_t>(CoreResource::BILLBOARD_DEFAULT_TEXTURE));
+
+	float vertices[20] =
+	{
+			0.5f,  0.5f, 0.0f,		1.0f, 1.0f,
+			0.5f, -0.5f, 0.0f,		1.0f, 0.0f,
+		   -0.5f, -0.5f, 0.0f,		0.0f, 0.0f,
+		   -0.5f,  0.5f, 0.0f,		0.0f, 1.0f
+	};
+	unsigned int indices[6] =
+	{
+		0, 1, 3,
+		1, 2, 3
+	};
+
+	glGenVertexArrays(1, &vao);
+	glGenBuffers(1, &vbo);
+	glGenBuffers(1, &ebo);
+
+	glBindVertexArray(vao);
+
+	glBindBuffer(GL_ARRAY_BUFFER, vbo);
+	glBufferData(GL_ARRAY_BUFFER, sizeof(vertices), vertices, GL_STATIC_DRAW);
+
+	glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, ebo);
+	glBufferData(GL_ELEMENT_ARRAY_BUFFER, sizeof(indices), indices, GL_STATIC_DRAW);
+
+	glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 5 * sizeof(float), (void*)0);
+	glEnableVertexAttribArray(0);
+
+	glVertexAttribPointer(1, 2, GL_FLOAT, GL_FALSE, 5 * sizeof(float), (void*)(3 * sizeof(float)));
+	glEnableVertexAttribArray(1);
+
+	glBindBuffer(GL_ARRAY_BUFFER, 0);
+	glBindVertexArray(0);
+}
+
+void ComponentBillboard::SwitchFrame()
+{
+	time_since_start += App->time->delta_time;
+	APP_LOG_INFO("%.1f", time_since_start);
+
+	if (time_since_start * sheet_speed >= 1)
+	{
+		current_sprite_x += 1;
+
+		if ((int)current_sprite_x >= x_tiles) {
```

Comment:
code style

Reply:
I don't find the problem in code style here, can you point it out?

Reply:
yeah you should open brackets on the next line not on the same one